### ISSUE TITLE
bug: `templ generate --watch --proxy` triggers 2 browser reload events

### ISSUE THREAD

**Issue by atipugin** [a-h/templ#891]
**Before you begin**
> Please make sure you're using the latest version of the templ CLI (`go install github.com/a-h/templ/cmd/templ@latest`), and have upgraded your project to use the latest version of the templ runtime (`go get -u github.com/a-h/templ@latest`)

Done.

**Describe the bug**
For some reason `templ generate --watch` processes same file multiple times on a single save. As result, it triggers several page reloads per change. I disabled all formatters to ensure there's nothing to emit another fs event, but it didn't help.

I've tried to reproduce it inside devcontainer (used `mcr.microsoft.com/devcontainers/go:1-1.22-bookworm`) and everything works just fine. So I guess the issue is with macOS FS events stuff.

**To Reproduce**
1. Start `templ generate --watch -v --proxy="http://localhost:8080" --cmd="go run cmd/myapp/main.go"` 
2. Change `.templ` file

**Expected behavior**
It should emit only one reload event.

**Screenshots**
<img width="1552" alt="SCR-20240819-opao" src="https://github.com/user-attachments/assets/79bb4d48-9801-472d-97c1-eb842285f0bb">

https://github.com/user-attachments/assets/db8c8ad0-76f0-4daf-a44b-eb9a6e94c465

**Logs**
None.

**`templ info` output**
```
usage: templ <command> [<args>...]

templ - build HTML UIs with Go

See docs at https://templ.guide

commands:
  generate   Generates Go code from templ files
  fmt        Formats templ files
  lsp        Starts a language server for templ files
  version    Prints the version
```

**Desktop (please complete the following information):**
 - OS: macOS Sonoma 14.6.1
 - templ CLI version: v0.2.747
- Go version (`go version`): go1.23.0 darwin/arm64
- `gopls` version (`gopls version`): v0.16.1

**Additional context**
None.

**Comment by joerdav** [comment]
I think the issue here is that templ updates the `.txt` file, but also is has a file watcher on that dir. And then triggers a reload as `textUpdated` is set to true: https://github.com/a-h/templ/blob/main/cmd/templ/generatecmd/eventhandler.go#L115

The only reason I can see that we would do this is if we were expecting that processes outside of templ to be able to update the dev mode files. @a-h My suggestion is that we do not return updated in this case so that text changes don't result in a double reload?

**Comment by a-h** [comment]
Yes, agreed, updated in PR #1027 (https://github.com/a-h/templ/pull/1027/commits/7553049adc0280504d40113a4c5aa05e1eacc5ce) this issue will be closed by #1027.

**Comment by a-h** [comment]
Fixed in merged PR https://github.com/a-h/templ/pull/1027
